### ISSUE TITLE
Train failed with multi gpus

### ISSUE THREAD

**Issue by stone002** [gojasper/flash-diffusion#18]
Hi, I meet this problem when I try to use 2 A100 to train.

I used train_flash_sdxl.py, my Trainer params are:
![image](https://github.com/user-attachments/assets/7446944d-54ee-4dce-a3f7-79b5a17ccd16)

and the flash_sdxl.yaml is 
![image](https://github.com/user-attachments/assets/8db84679-2460-483e-acf2-4ab731b8cb07)

My code stuck here every time.
![975c0938e966d62fbb9b748e42e073f](https://github.com/user-attachments/assets/8ff7a781-84c3-4ae7-9c56-4e9d960275f4)

If I use only one A100, it works fine, but very slow.

**Comment by LeonNerd** [comment]
hi, I have a question, when training sd3 flash, how should I create data, is the image in json the name of the image? It was also strange that I didn't have a model saved during my training

**Comment by stone002** [comment]
> hi, I have a question, when training sd3 flash, how should I create data, is the image in json the name of the image? It was also strange that I didn't have a model saved during my training

I only test sdxl, Flash Diffusion used 'webdataset' for training. I created my test dataset as the format:

**Comment by stone002** [comment]
> hi, I have a question, when training sd3 flash, how should I create data, is the image in json the name of the image? It was also strange that I didn't have a model saved during my training

I only test sdxl, Flash Diffusion used 'webdataset' for training. I created my test dataset as the format, one image with a same name json file, the json looks like:
`{
                         "jpg":img_name+".jpg",
                         "json":{
                             "caption":caption,
                             "aesthetic_score": 8.0,
                         }
                     }
`
then package all images and jsons in a .tar. It works fine.
About the model save, you may check your 'MAX_EPOCHS' and 'CKPT_EVERY_N_STEPS' in your .yaml file, or add 'save_last=True' in Trainner callbacks. Hope it helps!

**Comment by stone002** [comment]
Finally I found that because my test dataset is not enough. I only create one .tar to test. I tried to get more data and make more .tar files, it works fine on multi gpus.

**Comment by stone002** [comment]
> Thanks for your reply, I continue to try again, but there is still no change.my datasets is one image with a same name json file.

Maybe more data is needed I guess. My test .tar file contains 100 images and 100 json files, it works fine on single gpu.

**Comment by LeonNerd** [comment]
hi,I think it's very strange for me,but I can't figure out why.My json file is as follows：
{
    "jpg": "people_flux_00209.png",
    "json": {
        "caption": "A person within an Africa-shaped border with a tattooed arm and sleeveless top.",
        "aesthetic_score": 8
    }
}
and I packaged the image with the same name into a tar file as a training set with 3000 images.
# Dataset
SHARDS_PATH_OR_URLS:
  - pipe:cat /data/sd3/flash_sd3-{000000..000000}.tar
  But no loss is shown in the training, and no model is saved. My code is as follows
  trainer = Trainer(
        accelerator="cuda",
        devices=int(1),
        num_nodes=int(12),
#         devices=int(os.environ["SLURM_NPROCS"]) // int(os.environ["SLURM_NNODES"]),
#         num_nodes=int(os.environ["SLURM_NNODES"]),
        strategy="ddp_find_unused_parameters_true",
        default_root_dir=dir_path,
        max_epochs=args["MAX_EPOCHS"],
#         logger=loggers.WandbLogger(
#             project="flash-diffusion",
#             offline=False,
#             save_dir=dir_path,
#             name=run_name,
#         ),
        callbacks=[
#             WandbSampleLogger(log_batch_freq=args["LOG_EVERY_N_BATCHES"]),
            ModelCheckpoint(
                dirpath=ckpt_path,
                filename="{step}",
#                 every_n_train_steps=args["CKPT_EVERY_N_STEPS"],
                every_n_epochs = 1,
                save_top_k=-1,  # to save all the models
            ),
        ],
        num_sanity_val_steps=0,
        precision="bf16-mixed",
        check_val_every_n_epoch=100000000,
    )

    trainer.fit(pipeline, data_module)
![image](https://github.com/user-attachments/assets/c5a76231-8c39-42e0-8d35-8f618d91f06f)

**Comment by stone002** [comment]
> hi,I think it's very strange for me,but I can't figure out why.My json file is as follows： { "jpg": "people_flux_00209.png", "json": { "caption": "A person within an Africa-shaped border with a tattooed arm and sleeveless top.", "aesthetic_score": 8 } } and I packaged the image with the same name into a tar file as a training set with 3000 images.
> 
> # Dataset
> SHARDS_PATH_OR_URLS:
> 
> * pipe:cat /data/sd3/flash_sd3-{000000..000000}.tar
>   But no loss is shown in the training, and no model is saved. My code is as follows
> 
> ![image](https://private-user-images.githubusercontent.com/51089230/364273980-5370ff53-4bd0-4ee1-918a-f626c5b67758.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU0OTkyMzQsIm5iZiI6MTcyNTQ5ODkzNCwicGF0aCI6Ii81MTA4OTIzMC8zNjQyNzM5ODAtNTM3MGZmNTMtNGJkMC00ZWUxLTkxOGEtZjYyNmM1YjY3NzU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA1VDAxMTUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5ZmRlMmIwZGJjMTNiYjdlY2I1NTY5NjYxMjQ3YjkyZWZhZDllZmM5ODA1Yjg1ZWY4MDBlMDJmNzU4N2IyMjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bPw6AW2DMtp_3TlTd4XEq3NTv1Df5PerbXhQCid2ReQ)
> 
> ![image](https://private-user-images.githubusercontent.com/51089230/364271118-c5a76231-8c39-42e0-8d35-8f618d91f06f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU0OTkyMzQsIm5iZiI6MTcyNTQ5ODkzNCwicGF0aCI6Ii81MTA4OTIzMC8zNjQyNzExMTgtYzVhNzYyMzEtOGMzOS00MmUwLThkMzUtOGY2MThkOTFmMDZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA1VDAxMTUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkM2JlMjc2NGM1YTg2NzNhY2NiNTMyNzFkN2JiYWIwNDNhYjQyNjBlY2M1MTcwNGJkODAxMzA3MGQ5ZjI1OTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.e1VqCE1NiiQ3OX4OMf2AK-nccBAA6XFN1UpOAKsU0-A)

I used sdxl for test. I modified these in my code:
![image](https://github.com/user-attachments/assets/fdb388d4-a93c-4434-a7f0-306791293351)
![image](https://github.com/user-attachments/assets/a4d6578c-e1c4-4ced-8ef1-d149e575b4b3)

My test dataset is 100 [image, json] pairs, these values works for me but inference effect is not very good, I guess because my data and train epochs are both not enough. You can have a try by setting CKPT_EVERY_N_STEPS smaller to check if it saves .ckpt in training process. Hope it helps

**Comment by solitaryTian** [comment]
> Hi, I meet this problem when I try to use 2 A100 to train.
> 
> I used train_flash_sdxl.py, my Trainer params are: ![image](https://github.com/user-attachments/assets/7446944d-54ee-4dce-a3f7-79b5a17ccd16)
> 
> and the flash_sdxl.yaml is ![image](https://github.com/user-attachments/assets/8db84679-2460-483e-acf2-4ab731b8cb07)
> 
> My code stuck here every time. ![975c0938e966d62fbb9b748e42e073f](https://github.com/user-attachments/assets/8ff7a781-84c3-4ae7-9c56-4e9d960275f4)
> 
> If I use only one A100, it works fine, but very slow.

I have meet similar problem. Have you sloved it ? If so, can you share your solution?
